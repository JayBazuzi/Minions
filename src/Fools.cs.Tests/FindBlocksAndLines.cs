using ApprovalTests.Reporters;
using Fools.cs.AST;
using NUnit.Framework;

namespace Fools.cs.Tests
{
	[TestFixture, UseReporter(typeof(DiffReporter))]
	public class FindBlocksAndLines
	{
		[Test]
		public void should_find_nothing_in_an_empty_file()
		{
			string.Empty.find_blocks().should_parse_correctly();
		}

		[Test]
		public void file_with_a_single_empty_block_should_parse()
		{
			@"
some block:
	pass
".find_blocks().should_parse_correctly();
		}
	}
}
