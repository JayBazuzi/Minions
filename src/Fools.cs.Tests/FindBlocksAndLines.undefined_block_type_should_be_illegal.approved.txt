Should fail

/** Declarations **/
[]

/** Errors **/
[
  {
    "file_name": "fake file name.fool",
    "line": 1,
    "error_type": "Unrecognized statement",
    "expected": null,
    "actual": null,
    "snippet": "frazzle:",
    "suggestion": "I could not figure out this statement. Sorry that I can't give you a more useful error message. Perhaps you need to make Fools smarter?"
  },
  {
    "file_name": "fake file name.fool",
    "line": 1,
    "error_type": "Failed to parse 'program'.",
    "expected": null,
    "actual": null,
    "snippet": null,
    "suggestion": "Hopefully I also gave you some more specific error messages. Try fixing those first."
  }
]
